src/main/java/teammates/logic/api/Logic.java

```diff
@@ -2622,4 +2622,14 @@ public MimeMessage emailErrorReport(String path, String params, Throwable error)
     @SuppressWarnings("unused")
     private void ____helper_methods________________________________________() {
     }
+    
+    public void extractActiveAndArchivedCourses(List<CourseDetailsBundle> courseBundles, 
```

Comment:
@bamboo3250 actually, you can retrieve data from the database once (i.e. all courses), and extract the active and archived courses from that data set using two methods. That saves unnecessary database reads, but still achieves the same outcome.